### ISSUE TITLE
`export default interface` with two names should be an error

### ISSUE THREAD

**Issue by elibarzilay** [microsoft/TypeScript#39805]
The following does not report an error:

```ts
export default interface A { a: string; }
export default interface B { b: string; }
```

and instead makes both names refer to the merged interface.

---

Sidenote: if this does not become an error for whatever reason, then this should be made to work too:

```ts
export default interface A { a: string; }
export default interface B { b: string; }
declare var xa: A;
declare var xb: B;
```
